src/constructs/cloudwatch/lambda-alarms.ts

```diff
@@ -4,20 +4,14 @@ import type { GuLambdaFunction } from "../lambda";
 import type { GuAlarmProps } from "./alarm";
 import { GuAlarm } from "./alarm";
 
-export type LambdaMonitoring = NoMonitoring | ErrorPercentageMonitoring;
-
-export interface NoMonitoring {
-  noMonitoring: true;
-}
-
-export interface ErrorPercentageMonitoring
+export interface LambdaErrorPercentageMonitoring
   extends Omit<GuAlarmProps, "metric" | "threshold" | "comparisonOperator" | "evaluationPeriods"> {
   toleratedErrorPercentage: number;
   numberOfFiveMinutePeriodsToEvaluate?: number;
   noMonitoring?: false;
 }
 
-interface GuLambdaAlarmProps extends ErrorPercentageMonitoring {
+interface GuLambdaAlarmProps extends LambdaErrorPercentageMonitoring {
   lambda: GuLambdaFunction;
 }
```

Comment:
I agree that your suggestion would be an improvement. At the moment it’s tricky to do it this way because it means that any pattern which wants to create an alarm has to pass a `GuLambdaFunction` in as a prop. Since all of the patterns which use this construct are also creating the `GuLambdaFunction` itself, we end up with a weird circular dependency!

To work around this, I think we’d have to:

a) Define a separate interface that the lambda patterns can populate e.g.

```
interface GuLambdaMonitoringProps
  extends Omit<GuAlarmProps, "metric" | "threshold" | "comparisonOperator" | "evaluationPeriods"> {
  noMonitoring?: false;
}

// Library users create this object and pass it to the pattern
export interface GuLambdaErrorPercentageMonitoringProps extends GuLambdaMonitoringProps {
  toleratedErrorPercentage: number;
  numberOfFiveMinutePeriodsToEvaluate?: number;
}

// Patterns supplement the user-defined object with an additional lambda prop
interface GuLambdaErrorPercentageAlarmProps extends GuLambdaErrorPercentageMonitoringProps {
  lambda: GuLambdaFunction;
}
```

b) Relax [this linting rule](https://github.com/guardian/cdk/blob/main/eslint/rules/valid-constructors.js#L63-L70) so the patterns can pass the `GuLambdaFunction` in separately from the other props.

Reply:
Ah yes! I hadn't thought about that. In that case, I think how it is now is fine. Maybe it's something that can be refactored if and when we ever come to adding more lambda monitoring.